tools/src/main/scala/scala/scalanative/build/LLVM.scala

```diff
@@ -173,12 +173,39 @@ private[scalanative] object LLVM {
       case Success(_) =>
     }
 
+  /** This function allows a project to have multiple `main` files by copying
+   *  the one selected to the same parent directory as the `workDir` which is by
+   *  default named `native`. Since the directory is named `native`, having a
+   *  project named `native` will by default produce a executable named `native`
+   *  which will throw an exception since the copy command uses
+   *  REPLACE_EXISTING.
+   *
+   *  Having a project or `baseName` named `native` conflict with the build.
+   */
   private def copyOutput(config: Config, buildPath: Path) = {
     val outPath = config.artifactPath
-    config.compilerConfig.buildTarget match {
-      case BuildTarget.Application =>
-        Files.copy(buildPath, outPath, StandardCopyOption.REPLACE_EXISTING)
-      case _: BuildTarget.Library => outPath
+    try {
+      config.compilerConfig.buildTarget match {
+        case BuildTarget.Application =>
+          Files.copy(buildPath, outPath, StandardCopyOption.REPLACE_EXISTING)
+        case _: BuildTarget.Library => outPath
+      }
+    } catch {
+      case t: Throwable =>
```

Comment:
Typically we should not catch Throwable. Consider using Exception or IOException instead

Reply:
Probably going with IOException would be the best choice, there is no other Exception that `Files.copy` can throw. We also consider only having `case ex: IOExcpeiton if outPath.toFile().exists() => ` handler. In case of unexpected exception it might be better to pass it through so at least we'll get a stack trace (but it's unlikely to catch anything else here)

Reply:
Agree that this is a much better solution - this case we know but others are probably completely out of our control.

Besides, this looks nicer.